src/libs/PolicyUtils.ts

```diff
@@ -0,0 +1,217 @@
+        policy &&
+        policy?.isPolicyExpenseChatEnabled &&
+        policy?.role === CONST.POLICY.ROLE.ADMIN &&
+        (isOffline || policy?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || Object.keys(policy.errors ?? {}).length > 0)
+    );
+}
+
+function isExpensifyTeam(email: string): boolean {
+    const emailDomain = Str.extractEmailDomain(email ?? '');
+    return emailDomain === CONST.EXPENSIFY_PARTNER_NAME || emailDomain === CONST.EMAIL.GUIDES_DOMAIN;
+}
+
+function isExpensifyGuideTeam(email: string): boolean {
+    const emailDomain = Str.extractEmailDomain(email ?? '');
+    return emailDomain === CONST.EMAIL.GUIDES_DOMAIN;
+}
+
+/**
+ * Checks if the current user is an admin of the policy.
+ */
+const isPolicyAdmin = (policy: OnyxTypes.Policy): boolean => policy?.role === CONST.POLICY.ROLE.ADMIN;
+
+/**
+ * Create an object mapping member emails to their accountIDs. Filter for members without errors, and get the login email from the personalDetail object using the accountID.
+ *
+ * We only return members without errors. Otherwise, the members with errors would immediately be removed before the user has a chance to read the error.
+ */
+function getMemberAccountIDsForWorkspace(policyMembers: PolicyMemberList, personalDetails: PersonalDetailsList): MemberEmailsToAccountIDs {
+    const memberEmailsToAccountIDs: Record<string, string> = {};
+    Object.keys(policyMembers ?? {}).forEach((accountID) => {
+        const member = policyMembers?.[accountID];
+        if (Object.keys(member?.errors ?? {})?.length > 0) {
+            return;
+        }
+        const personalDetail = personalDetails[accountID];
+        if (!personalDetail?.login) {
+            return;
+        }
+        memberEmailsToAccountIDs[personalDetail.login] = accountID;
+    });
+    return memberEmailsToAccountIDs;
+}
+
+/**
+ * Get login list that we should not show in the workspace invite options
+ */
+function getIneligibleInvitees(policyMembers: PolicyMemberList, personalDetails: PersonalDetailsList): string[] {
+    const memberEmailsToExclude: string[] = [...CONST.EXPENSIFY_EMAILS];
+    Object.keys(policyMembers ?? {}).forEach((accountID) => {
+        const policyMember = policyMembers?.[accountID];
+        // Policy members that are pending delete or have errors are not valid and we should show them in the invite options (don't exclude them).
+        if (policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || Object.keys(policyMember?.errors ?? {}).length > 0) {
+            return;
+        }
+        const memberEmail = personalDetails?.[accountID]?.login;
+        if (!memberEmail) {
+            return;
+        }
+        memberEmailsToExclude.push(memberEmail);
+    });
+
+    return memberEmailsToExclude;
+}
+
+/**
+ * Gets the tag from policy tags, defaults to the first if no key is provided.
+ */
+function getTag(policyTags: Record<string, OnyxTypes.PolicyTag>, tagKey?: keyof typeof policyTags) {
```

Comment:
```suggestion
function getTag(policyTags: OnyxEntry<OnyxTypes.PolicyTags>, tagKey?: keyof typeof policyTags) {
```

What about creating a `PolicyTags` type like I suggested for `PolicyMembers` and using it here?

Reply:
Add return type for the function